src/beamlime/applications/daemons.py

```diff
@@ -1,72 +1,63 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
-import json
+import asyncio
 import os
 from dataclasses import dataclass
-from typing import AsyncGenerator, Generator, Union
+from typing import AsyncGenerator, NewType, Union
 
-from scippneutron.io.nexus.load_nexus import JSONGroup, json_nexus_group
-
-from ._parameters import ChunkSize, DataFeedingSpeed
-from ._random_data_providers import RandomEvents
+from ._nexus_helpers import NexusContainer
+from ._parameters import DataFeedingSpeed, NumFrames
+from ._random_data_providers import DetectorNumberCands, random_ev44_generator
 from .base import Application, DaemonInterface, MessageProtocol
-from .handlers import Events
 
 Path = Union[str, bytes, os.PathLike]
 
 
 @dataclass
 class RunStart:
-    content: JSONGroup
+    content: NexusContainer
 
 
 @dataclass
-class RawDataSent:
-    content: Events
+class DetectorDataReceived:
+    content: dict
+
 
+NexusTemplatePath = NewType("NexusTemplatePath", str)
 
-class DataStreamSimulator(DaemonInterface):
-    """Data that simulates the data streaming from the random generator."""
 
-    random_events: RandomEvents
-    chunk_size: ChunkSize
-    data_feeding_speed: DataFeedingSpeed
+class ESSKafkaStreamSimulator(DaemonInterface):
```

Comment:
This name is confusing. The fake that we are implementing is meant to stand in for the Kafka stream listener. It does not do anything with the Kafka side of things (and as explained on Tuesday, we may later have a `FakeKafka`, so we can test the "real" listener. What was wrong with `FakeListener`? Sure it has to "produce" something that it forwards, but it is "faking" the real listener (which will be implemented later).

Reply:
I don't know... I agree that it should not include `Kafka` but `Listener` is already 
but `FakeListener` sounds like a stub or a mock rather than something actually has a functionality...
But name doesn't really matter... I can revert the name

Reply:
A "fake" is generally more than a simple stub or mock. Here is a ChatGPT summary:

```
In software development and testing, stubs, mocks, and fakes are all used to simulate parts of a system that are not yet available or to isolate components for testing purposes. While they serve similar purposes, they have different characteristics and usage contexts:

1. **Stub**:
   - A stub is a minimal implementation of a component or module that provides canned responses to function calls or method invocations.
   - Stubs are typically used in integration testing when a component depends on another component that is not yet implemented or available.
   - Stubs do not have full functionality and are often hardcoded to return specific values or execute predefined behavior.
   - They are simple placeholders that allow developers to continue working on parts of the system independently.

2. **Mock**:
   - A mock is a test double that mimics the behavior of a real object or component in a controlled way.
   - Mocks are used primarily in unit testing to verify interactions between objects and to test the behavior of the object being tested.
   - Unlike stubs, mocks are programmed with expectations about how they should be called (i.e., which methods and with what parameters).
   - Mocks can simulate both expected and unexpected behaviors, allowing developers to test edge cases and error handling.

3. **Fake**:
   - A fake is a simplified functional implementation of a dependency that is used in place of the actual implementation.
   - Fakes are often used in testing when the real implementation is too complex, slow, or resource-intensive to use in testing scenarios.
   - Fakes are more sophisticated than stubs because they provide a functional alternative, but they are simpler than the actual implementation.
   - For example, a fake database implementation might store data in memory rather than on disk to speed up testing.

In summary, stubs provide basic, predefined responses to method calls, mocks are used to verify interactions and behavior during unit testing, and fakes provide functional but simplified alternatives to real dependencies for testing purposes. Each serves a different purpose in software development and testing, helping developers write robust and maintainable code.
```